src/theory/quantifiers/sygus/sygus_pbe.h

```diff
@@ -322,37 +322,67 @@ class CegConjecturePbe : public SygusModule
   //--------------------------------- end PBE search values
 
   // -------------------------------- decision tree learning
-  // index filter
-  class IndexFilter {
-  public:
-    IndexFilter(){}
-    void mk( std::vector< Node >& vals, bool pol = true );
-    std::map< unsigned, unsigned > d_next;
-    unsigned start();
-    unsigned next( unsigned i );
-    void clear() { d_next.clear(); }
-    bool isEq( std::vector< Node >& vs, Node v );
-  };
-  // subsumption trie
-  class SubsumeTrie {
-  public:
-    SubsumeTrie(){}
-    // adds term to the trie, removes based on subsumption
-    Node addTerm( CegConjecturePbe * pbe, Node t, std::vector< Node >& vals, bool pol, std::vector< Node >& subsumed, IndexFilter * f = NULL );
-    // adds condition to the trie (does not do subsumption)
-    Node addCond( CegConjecturePbe * pbe, Node c, std::vector< Node >& vals, bool pol, IndexFilter * f = NULL );
-    // returns the set of terms that are subsets of vals
-    void getSubsumed( CegConjecturePbe * pbe, std::vector< Node >& vals, bool pol, std::vector< Node >& subsumed, IndexFilter * f = NULL );
-    // returns the set of terms that are supersets of vals
-    void getSubsumedBy( CegConjecturePbe * pbe, std::vector< Node >& vals, bool pol, std::vector< Node >& subsumed_by, IndexFilter * f = NULL );
-    // v[-1,1,0] -> children always false, always true, both
-    void getLeaves( CegConjecturePbe * pbe, std::vector< Node >& vals, bool pol, std::map< int, std::vector< Node > >& v, IndexFilter * f = NULL );
+  /** Subsumption trie
+  *
+  * This class manages a set of terms for a PBE sygus enumerator.
+  *
+  * In PBE sygus, we are interested, for each term t, the set of I/O examples
+  * that it satisfies, which can be represented by a vector of Booleans.
+  * For example, given conjecture:
+  *   f( 1 ) = 2 ^ f( 3 ) = 4 ^ f( -1 ) = 1 ^ f( 5 ) = 5
+  *   For term x satisfies 0001,
```

Comment:
Clarify that "x" is the term "f" is interpreted on and that "0001" is because only the fifth example works. Also this "for term x satisfies" is strange to parse.